ndk-build/src/apk.rs

```diff
@@ -155,6 +160,41 @@ impl<'a> UnalignedApk<'a> {
         Ok(())
     }
 
+    pub fn add_file(&self, base_path: &Path, path: &Path) -> Result<(), NdkError> {
+        let out = self.0.build_dir.join(path);
+        let out_dir = out.parent().unwrap();
+        std::fs::create_dir_all(&out_dir)?;
+        std::fs::copy(base_path.join(path), out)?;
+
+        let mut aapt = self.0.build_tool(bin!("aapt"))?;
+        aapt.arg("add").arg(self.0.unaligned_apk()).arg(path);
+        if !aapt.status()?.success() {
+            return Err(NdkError::CmdFailed(aapt));
+        }
+        Ok(())
+    }
+
+    fn add_entry(&self, base_path: &Path, path: &Path) -> Result<(), NdkError> {
+        let full_path = base_path.join(path);
+        if full_path.is_dir() {
+            for entry in fs::read_dir(full_path)? {
```

Comment:
Can't we pass a directory to `aapt add` so that it adds them recursively?

If not you're better off building an array of filenames and adding them to the archive in one go, spawning an `aapt add` process per file can potentially become very costly and wasteful.